cdap-unit-test/src/main/java/co/cask/cdap/test/TestBase.java

```diff
@@ -171,9 +171,14 @@ protected ApplicationManager deployApplication(Class<? extends Application> appl
    *
    * @deprecated Use {@link TestManager#clear()} from {@link #getTestManager()}.
    */
-  protected void clear() throws Exception {
-    TestManager testManager = getTestManager();
-    testManager.clear();
+  protected void clear() {
+    try {
+      TestManager testManager = getTestManager();
+      testManager.clear();
+    } catch (Exception e) {
+      // Unchecked exception to maintain compatibility until we remove this method
+      Throwables.propagate(e);
```

Comment:
https://code.google.com/p/guava-libraries/wiki/ThrowablesExplained

"... but throwing the result -- e.g. throw Throwables.propagate(t) -- can be useful to prove to the compiler that an exception will be thrown."